dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserSearch/NamesSynonymService.cs

```diff
@@ -0,0 +1,55 @@
+namespace TeacherIdentity.AuthServer.Services.UserSearch;
+
+public class NamesSynonymService : INamesSynonymService
+{
+    private readonly Dictionary<string, HashSet<string>> namesLookup = new Dictionary<string, HashSet<string>>();
+
+    public NamesSynonymService()
+    {
+        Initialise();
+    }
+
+    public IList<string> GetSynonyms(string name)
+    {
+        if (namesLookup.TryGetValue(name.ToLower(), out var synonyms))
+        {
+            return synonyms.ToList();
+        }
+
+        return new List<string>();
+    }
+
+    private void Initialise()
+    {
+        var namesFilePath = Path.Combine(AppContext.BaseDirectory, "names.csv");
+        using var textReader = File.OpenText(namesFilePath);
+        string? line = null;
+        while ((line = textReader.ReadLine()) != null)
+        {
+            if (line.Length == 0 || line[0] == '#')
+            {
+                continue; // ignore empty lines and comments
+            }
+
+            var names = line.Split(',');
+            foreach (var name in names)
+            {
+                HashSet<string>? synonyms;
+                if (!namesLookup.TryGetValue(name, out synonyms))
+                {
+                    synonyms = new HashSet<string>();
```

Comment:
Does this need to be a `HashSet<>` if we're not exposing it as such?

Reply:
I was using the fact that if you are adding something that is already in the list it will ignore it and not add it again.
What was your alternative suggestion ?

Reply:
If there can be duplicates then `HashSet<>` makes sense. Presumably we don't need to worry about case sensitivity? Using a regular `List<>` was my thinking.